app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,34 @@
+## TODO: count param for seeing end of a short query
+
+class RecipesController < ApplicationController
+	MAX_PER_PAGE = 12
+
+	def index
+		@query = params[:q]
+		@from = params[:from].nil? ? 0 : params[:from].to_i
+		@to = params[:to].nil? ? @from + MAX_PER_PAGE : params[:to].to_i
+
+		@recipes = RecipeApiWrapper.list_recipes(@query, @from, @to)
+
```

Comment:
If you're expecting your lib file to raise an error, you should wrap this call in a `begin`/`rescue`.